tokenizers/src/models/mod.rs

```diff
@@ -68,6 +72,83 @@ pub enum ModelWrapper {
+        struct ModelWrapperVisitor;
+
+        impl<'de> Visitor<'de> for ModelWrapperVisitor {
+            type Value = ModelWrapper;
+
+            fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+                formatter.write_str("a valid ModelWrapper")
+            }
+
+            fn visit_map<V>(self, mut map: V) -> std::result::Result<ModelWrapper, V::Error>
+            where
+                V: MapAccess<'de>,
+            {
+                println!("Parsed stuff????");
+
+                // Look for a "type" tag
+                if let Some(key) = map.next_key::<String>()? {
+                    println!("Matched key: {:?}", key);
+                    match key.as_str() {
+                        "type" => {
+                            let tag: String = map.next_value()?;
+                            println!("Matched tag: {:?}", tag);
+
+                            match tag.as_str() {
+                                "BPE" => {
+                                    BPE::deserialize(MapAccessDeserializer::new(map))?;
+                                }
+                                "WordPiece" => {
+                                    WordPiece::deserialize(MapAccessDeserializer::new(map))?;
+                                }
+                                "WordLevel" => {
+                                    WordLevel::deserialize(MapAccessDeserializer::new(map))?;
+                                }
+                                "Unigram" => {
+                                    Unigram::deserialize(MapAccessDeserializer::new(map))?;
+                                }
+                                _ => {
+                                    return Err(V::Error::unknown_variant(
+                                        &tag,
+                                        &["BPE", "WordPiece", "WordLevel", "Unigram"],
+                                    ));
+                                }
+                            }
+                        }
+                        _ => {
+                            #[derive(Deserialize, Debug, PartialEq, Clone)]
+                            #[serde(untagged)]
+                            pub enum DummyModelWrapper {
+                                BPE(BPE),
+                                // WordPiece must stay before WordLevel here for deserialization (for retrocompatibility
+                                // with the versions not including the "type"), since WordLevel is a subset of WordPiece
+                                WordPiece(WordPiece),
+                                WordLevel(WordLevel),
+                                Unigram(Unigram),
+                            }
+
+                            match DummyModelWrapper::deserialize(MapAccessDeserializer::new(map))? {
+                                DummyModelWrapper::BPE(inner) => ModelWrapper::BPE(inner),
+                                _ => todo!(),
```

Comment:
this is the way to default to the automatically derived deserialize, instead of doing it ourlseved